src/Core/Checkout/Customer/Aggregate/CustomerGroup/CustomerGroupValidator.php

```diff
@@ -0,0 +1,79 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Customer\Aggregate\CustomerGroup;
+
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\DataAbstractionLayer\Write\Command\CascadeDeleteCommand;
+use Shopware\Core\Framework\DataAbstractionLayer\Write\Command\DeleteCommand;
+use Shopware\Core\Framework\DataAbstractionLayer\Write\Validation\PreWriteValidationEvent;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\Framework\Validation\WriteConstraintViolationException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Validator\ConstraintViolation;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\ConstraintViolationList;
+
+class CustomerGroupValidator implements EventSubscriberInterface
+{
+    public const VIOLATION_DELETE_DEFAULT_CUSTOMER_GROUP = 'delete_default_customer_group_violation';
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            PreWriteValidationEvent::class => 'preValidate',
+        ];
+    }
+
+    public function preValidate(PreWriteValidationEvent $event): void
+    {
+        $commands = $event->getCommands();
+
+        foreach ($commands as $command) {
+            $violations = new ConstraintViolationList();
+
+            if ($command instanceof CascadeDeleteCommand || $command->getDefinition()->getClass() !== CustomerGroupDefinition::class) {
+                continue;
+            }
+
+            $pk = $command->getPrimaryKey();
+            $id = \mb_strtolower(Uuid::fromBytesToHex($pk['id']));
+
+            if ($command instanceof DeleteCommand && $id === Defaults::LANGUAGE_SYSTEM) {
```

Comment:
Hey @JoshuaBehrens , please use an early return here. Also, please check whether `Defaults::LANGUAGE_SYSTEM` is actually what you intend to check for :wink:

Reply:
Thanks for the 📋 🍝 hint